src/api/cvc4cpp.cpp

```diff
@@ -3659,6 +3660,9 @@ void Grammar::addRule(const Term& ntSymbol, const Term& rule)
          "predeclaration";
   CVC4_API_CHECK(ntSymbol.d_node->getType() == rule.d_node->getType())
       << "Expected ntSymbol and rule to have the same sort";
+  CVC4_API_ARG_CHECK_EXPECTED(!containsFreeVariables(rule), rule)
+      << "a constant term or a term containing synthFun/synthInv parameters "
+         "and non-terminal symbols of the grammar";
```

Comment:
```suggestion
      << "Expected a term whose free variables are limited to synthFun/synthInv parameters "
         "and non-terminal symbols of the grammar";
```

Reply:
@ajreynol `Expected` is not necessary, it's part of the message of the macro.